app/models/interactive_page.rb

```diff
@@ -307,7 +313,25 @@ def self.import(page_json_object, helper=nil)
       page_json_object[:embeddables].each do |embed_hash|
         helper.set_references(embed_hash[:embeddable])
       end
+      # For older export files, if page intro exists, add it as a new embeddable in header_block
+      import_legacy_intro(import_page, page_json_object[:text])
     end
     import_page
   end
+
+  def self.import_legacy_intro(import_page, intro_text)
+    if intro_text
+      helper = LaraSerializationHelper.new if helper.nil?
+      intro_embeddable_hash = {
+                         "content": intro_text,
+                         "is_full_width": true,
+                         "is_hidden": false,
+                         "name": "",
+                         "type": "Embeddable::Xhtml"
+                        }
+      intro_embeddable = helper.import(intro_embeddable_hash)
+      import_page.show_header = true
```

Comment:
I'm not sure this ever happens, but it seems possible the old show_introduction could be false. And when that content is then imported show_header will be true.

I don't think this is worth fixing though. As far as I know we don't need to fully support importing legacy content.